src/ploomber/cli/examples.py

```diff
@@ -259,11 +261,24 @@ def list(self):
         tw.write('Example: ploomber examples -n templates/ml-basic -o ml\n\n')
 
     def download(self, name, output):
+        with open(self.examples / '_index.csv',
+                  newline='',
+                  encoding='utf-8-sig') as f:
+            rows = list(csv.DictReader(f))
         selected = self.path_to(name)
-
+        categories = []
+        for row in rows:
+            category = row.pop('name')
+            del row['idx']
+            categories.append(category)
```

Comment:
I think let's first rename categories to `names` (I don't what I was thinking when I named that variable) 

looks like all of this is repeated in the `list` method. I think let's move all of this to the class constructor and assign the `names` (previously named `categories`) to an instance variable

```python
self._names = names
```

then, we can re-use `self._names` in `list` and `download`

Reply:
How about the `name` arg in the function signature? Could this be a bit confusing?
`    def download(self, name, output):`

Reply:
let's rename categories to `example_names` then

Reply:
I was trying to merge them into the class constructor but just realized that they are different. 

* In `list()`: It's a dict (key: category, value: examples in the category), which may be the reason why it is named `category`.
```python
        by_category = defaultdict(lambda: [])
        for row in rows:
            category = row.pop('category')
            del row['idx']
            by_category[category].append(row)
```
* In `download()`, it's a list of all examples:
```python
        categories = []
        for row in rows:
            category = row.pop('name')
            del row['idx']
            categories.append(category)
```
Maybe We can just keep the former `by_category` dict and move it to the constructor, and we can use it in `list()`. Also, maybe we can rename `by_category` to `examples_by_category`?

For `download()`, looks like we need another variable `self._example_names = [example["name"] for category, examples in self._examples_by_category.items() for example in examples]` in the constructor.

Reply:
After finishing the above change, the `test_clones_in_home_directory` and `test_change_default_branch` tests will fail. Looks like the constructor is not called. How should we call the constructor? `monkeypatch.setattr(examples._ExamplesManager, '__init__', lambda _: None)` does not work.

https://github.com/qixuan27/ploomber/blob/master/tests/cli/test_examples.py#L104-L113